File: tests/fixtures/lang_smoke/python/shapes.py
"""Minimal Python fixture exercising CONTAINS (class -> method), a CALLS edge,
and a parameter used downstream (data_flow)."""


class Rectangle:
    """A rectangle shape with a width and height."""

    def __init__(self, width: float, height: float) -> None:
        self.width = width
        self.height = height

    def area(self) -> float:
        # method on a class -> CONTAINS edge (Rectangle contains area)
        return self.width * self.height

    def scaled_area(self, factor: float) -> float:
        # `factor` (a parameter) flows into the multiply -> data_flow.
        # self.area() is a method call -> CALLS edge.
        base = self.area()
        return base * factor

File: tests/fixtures/lang_smoke/javascript/shapes.js
// Minimal JavaScript fixture: class -> method (CONTAINS), method call (CALLS),
// and a parameter used downstream (data_flow).

class Rectangle {
  constructor(width, height) {
    this.width = width;
    this.height = height;
  }

  area() {
    // method on a class -> CONTAINS edge
    return this.width * this.height;
  }

  scaledArea(factor) {
    // `factor` (parameter) flows into the multiply -> data_flow;
    // this.area() is a method call -> CALLS edge.
    const base = this.area();
    return base * factor;
  }
}

module.exports = { Rectangle };
